src/code/z_camera.c

```diff
@@ -4398,101 +4400,136 @@ s32 Camera_Subj4(Camera* camera) {
-    at->y = eyeNext->y + rwData->unk_00.dir.y;
-    at->z = eyeNext->z + rwData->unk_00.dir.z;
+    Actor_GetWorldPosShapeRot(&playerPosRot, &camera->player->actor);
+    Math3D_LineClosestToPoint(&rwData->crawlspaceLine, &playerPosRot.pos, eyeNext);
+
+    // *at is unused before getting overwritten later this function
+    at->x = eyeNext->x + rwData->crawlspaceLine.dir.x;
+    at->y = eyeNext->y + rwData->crawlspaceLine.dir.y;
+    at->z = eyeNext->z + rwData->crawlspaceLine.dir.z;
+
     *eye = *eyeNext;
-    sp64.yaw = rwData->unk_30;
-    sp64.r = 5.0f;
-    sp64.pitch = 0x238C;
-    Camera_AddVecGeoToVec3f(&sp98, eyeNext, &sp64);
-    rwData->unk_2C += 0xBB8;
-    temp_f16 = Math_CosS(rwData->unk_2C);
-    eye->x += (sp98.x - eye->x) * fabsf(temp_f16);
-    eye->y += (sp98.y - eye->y) * fabsf(temp_f16);
-    eye->z += (sp98.z - eye->z) * fabsf(temp_f16);
-
-    if ((rwData->unk_28 < temp_f16) && !rwData->unk_2E) {
+
+    targetOffset.yaw = rwData->forwardYaw;
+    targetOffset.r = 5.0f;
+    targetOffset.pitch = 0x238C; // ~50 degrees
+
+    Camera_AddVecGeoToVec3f(&vEyeTarget, eyeNext, &targetOffset);
+
+    rwData->eyeLerpPhase += 0xBB8;
+    eyeLerp = Math_CosS(rwData->eyeLerpPhase);
+
+    // VEC3F_LERPIMPDST(eye, eye, &vEyeTarget, fabsf(eyeLerp))
+    eye->x += (vEyeTarget.x - eye->x) * fabsf(eyeLerp);
+    eye->y += (vEyeTarget.y - eye->y) * fabsf(eyeLerp);
+    eye->z += (vEyeTarget.z - eye->z) * fabsf(eyeLerp);
+
+    // When camera reaches the peak of offset and starts to move down
+    // && alternating cycles (sfx plays only every 2nd cycle)
+    if ((eyeLerp > rwData->eyeLerp) && !rwData->isSfxOff) {
         player = camera->player;
-        rwData->unk_2E = true;
-        func_800F4010(&player->actor.projectedPos, player->unk_89E + 0x8B0, 4.0f);
-    } else if (rwData->unk_28 > temp_f16) {
-        rwData->unk_2E = false;
+        rwData->isSfxOff = true;
+        func_800F4010(&player->actor.projectedPos, NA_SE_PL_CRAWL + player->unk_89E, 4.0f);
+    } else if (eyeLerp < rwData->eyeLerp) {
+        rwData->isSfxOff = false;
     }
 
-    rwData->unk_28 = temp_f16;
+    rwData->eyeLerp = eyeLerp;
+
     camera->player->actor.world.pos = *eyeNext;
     camera->player->actor.world.pos.y = camera->playerGroundY;
-    camera->player->actor.shape.rot.y = sp64.yaw;
-    temp_f16 = ((240.0f * temp_f16) * (rwData->unk_24 * 0.416667f));
-    temp_a0 = temp_f16 + rwData->unk_30;
-    at->x = eye->x + (Math_SinS(temp_a0) * 10.0f);
+    camera->player->actor.shape.rot.y = targetOffset.yaw;
+
+    eyeLerp = ((240.0f * eyeLerp) * (rwData->xzSpeed * 0.416667f));
+    eyeToAtYaw = rwData->forwardYaw + eyeLerp;
+
+    at->x = eye->x + (Math_SinS(eyeToAtYaw) * 10.0f);
     at->y = eye->y;
-    at->z = eye->z + (Math_CosS(temp_a0) * 10.0f);
+    at->z = eye->z + (Math_CosS(eyeToAtYaw) * 10.0f);
+
     camera->roll = Camera_LERPCeilS(0, camera->roll, 0.5f, 0xA);
-    return 1;
+
+    return true;
 }
 
+#undef vEyeTarget
+#undef vZoomTimer
```

Comment:
Same here, feel it would be better in the function. Commenting which temp they correspond to would also be nice, like you did for `vCrawlSpaceBackPos` and `vPlayerDistToFront` when you swapped the defines.